Example/Tests/PURLoggerStandardPluginTest.swift

```diff
@@ -0,0 +1,240 @@
+        XCTAssertTrue(logStorageContent.contains("{buffered.c|aaa:2}"))
+        XCTAssertTrue(logStorageContent.contains("{buffered.d|aaa:3}"))
+
+        XCTAssertFalse(logStorageContent.contains("{buffered.d|aaa:4}"))
+        XCTAssertFalse(logStorageContent.contains("{buffered.c|aaa:5}"))
+        XCTAssertFalse(logStorageContent.contains("{buffered.c|aaa:6}"))
+
+        logger.logStore().clearAll()
+        logger.shutdown()
+    }
+
+    func testBufferedOutputPlugin_periodicalFlushing() {
+        let loggerConfiguration = PURLoggerStandardPluginTest.loggerConfiguration
+        let logger = PURLogger(configuration: loggerConfiguration)
+        let testLogStorage = loggerConfiguration.logStorage
+
+        expectation(forNotification: Notification.Name.PURBufferedOutputDidStart.rawValue, object: nil, handler: nil)
+        waitForExpectations(timeout: 1.0, handler: nil)
+
+        XCTAssertEqual(testLogStorage.description, "")
+
+        expectation(forNotification: Notification.Name.PURBufferedOutputDidSuccessWriteChunk.rawValue, object: nil, handler: nil)
+
+        logger.post(["aaa": "1"], tag: "buffered.e")
+        logger.post(["aaa": "2"], tag: "buffered.e")
+        logger.post(["aaa": "3"], tag: "buffered.f")
+
+        XCTAssertEqual(testLogStorage.description, "")
+
+        // wait flush interval(2sec) ...
+        waitForExpectations(timeout: 3.0, handler: nil)
+
+        let logStorageContent = testLogStorage.description
+        XCTAssertTrue(logStorageContent.contains("{buffered.e|aaa:1}"))
+        XCTAssertTrue(logStorageContent.contains("{buffered.e|aaa:2}"))
+        XCTAssertTrue(logStorageContent.contains("{buffered.f|aaa:3}"))
+
+        logger.logStore().clearAll()
+        logger.shutdown()
+    }
+
+    func testBufferedOutputPlugin_retry() {
+        let loggerConfiguration = PURLoggerStandardPluginTest.loggerConfiguration
+        let logger = PURLogger(configuration: loggerConfiguration)
+        let testLogStorage = loggerConfiguration.logStorage
+
+        expectation(forNotification: Notification.Name.PURBufferedOutputDidStart.rawValue, object: nil, handler: nil)
+        waitForExpectations(timeout: 1.0, handler: nil)
+
+        XCTAssertEqual(testLogStorage.description, "")
+
+        expectation(forNotification: Notification.Name.PURBufferedOutputDidTryWriteChunk.rawValue, object: nil, handler: nil)
+
+        logger.post(["aaa": "1"], tag: "failure")
+        logger.post(["aaa": "2"], tag: "failure")
+        logger.post(["aaa": "3"], tag: "failure")
+        logger.post(["aaa": "4"], tag: "failure")
+        logger.post(["aaa": "5"], tag: "failure")
+
+        waitForExpectations(timeout: 1.0, handler: nil)
+        XCTAssertEqual(testLogStorage.description, "[error]")
+
+        expectation(forNotification: Notification.Name.PURBufferedOutputDidTryWriteChunk.rawValue, object: nil, handler: nil)
+        // scheduled after 2sec
+        waitForExpectations(timeout: 3.0, handler: nil)
```

Comment:
BufferedOutput schedules retry after 2s, 4s, 8s.
So test wait +1 seconds for every retries.